### DISCUSSION TITLE
Resource discovery and metric scraper version compatibility

### DISCUSSION THREAD

**Discussion by SurveySky** [tomkerkhove/promitor#2065] [Q&A]
Just wanted to get a clarification about resource discovery and metric scraper version compatibility if there is something like that. I have been testing versions and have noticed that sometimes metric scraper does not return metrics even though configuration is correct. As you have released new versions of both scraper and discovery I have noticed that using resource discovery helm version 0.8.0 with metric scraper helm version 2.8.0 is not working. No error on the configuration or any other side, just don't see any metrics from scraper. I have made it work with other versions, but then I am troubled by high resource usage by scrapers so I would love to use the new versions. So my question is: Could you make and publish some sort of compatibility table? And maybe you have some tips on why it does not work?

**Comment by tomkerkhove** [comment]
> As you have released new versions of both scraper and discovery I have noticed that using resource discovery helm version 0.8.0 with metric scraper helm version 2.8.0 is not working.

Can you elaborate on that please? What did you see? They should work nicely together.

**Reply by SurveySky** [reply]
Yeah sorry about that, I see that today it is working fine. Yesterday for some reason when exposing a pod I did not see any scraped metrics, but today everything seems to be working. But I have another question, when looking at scraper logs I see that it is showing an error:
![image](https://user-images.githubusercontent.com/91472898/176881329-c21d4443-0ff3-41c6-871a-c4d207f913de.png)
Why is this happening?

**Reply by tomkerkhove** [reply]
That might be due to https://github.com/tomkerkhove/promitor/pull/2050 - Would you mind sharing the full log please?
